feature/experimental/system/health/tests/system_generic_health_check/system_generic_health_check_test.go

```diff
@@ -0,0 +1,745 @@
+		}
+	}
+}
+
+func TestGetInterfaceName(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	dutPorts := sortPorts(dut.Ports())
+
+	for _, port := range dutPorts {
+		interfaces = append(interfaces, port.Name())
+	}
+	t.Logf("Interface: %s", interfaces)
+}
+
+func TestNoQueueDrop(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+
+	type testCase struct {
+		desc     string
+		path     string
+		counters []*ygnmi.Value[uint64]
+	}
+
+	for _, intf := range interfaces {
+		qosInterface := gnmi.OC().Qos().Interface(intf)
+		cases := []testCase{
+			{
+				desc:     "Queue Input Dropped packets",
+				path:     "/qos/interfaces/interface/input/queues/queue/state/dropped-pkts",
+				counters: gnmi.LookupAll(t, dut, qosInterface.Input().QueueAny().DroppedPkts().State()),
+			},
+			{
+				desc:     "Queue Output Dropped packets",
+				path:     "/qos/interfaces/interface/output/queues/queue/state/dropped-pkts",
+				counters: gnmi.LookupAll(t, dut, qosInterface.Output().QueueAny().DroppedPkts().State()),
+			},
+			{
+				desc:     "Queue input voq-output-interface dropped packets",
+				path:     "/qos/interfaces/interface/input/virtual-output-queues/voq-interface/queues/queue/state/dropped-pkts",
+				counters: gnmi.LookupAll(t, dut, qosInterface.Input().VoqInterfaceAny().QueueAny().DroppedPkts().State()),
+			},
+		}
+
+		for _, c := range cases {
+			desc := c.desc
+			counters := c.counters
+
+			t.Run(desc, func(t *testing.T) {
+				if len(counters) == 0 {
+					t.Skipf("%s Interface %s Telemetry Value is not present", desc, intf)
+				}
+				for queueID, dropPkt := range counters {
+					dropCount, present := dropPkt.Val()
+					if !present {
+						t.Errorf("%s Interface %s %s Telemetry Value is not present", desc, intf, dropPkt.Path)
+					} else {
+						if present && dropCount > 0 {
+							t.Errorf("%s Interface %s, Queue %d has %d drop(s)", desc, intf, queueID, dropCount)
+						}
+					}
+				}
+			})
+		}
+	}
+}
+
+func TestNoAsicDrop(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+
+	query := gnmi.OC().ComponentAny().IntegratedCircuit().PipelineCounters().Drop().State()
+	asicDrops := gnmi.GetAll(t, dut, query)
+	if len(asicDrops) == 0 {
+		t.Fatalf("ERROR: %s is not present", query)
+	}
+
+	for _, drop := range asicDrops {
+
+		if drop.InterfaceBlock != nil {
+
+			if drop.InterfaceBlock.InDrops != nil && drop.InterfaceBlock.GetInDrops() > 0 {
+				t.Errorf("ERROR: InDrops counter got %d, want 0", drop.InterfaceBlock.GetInDrops())
+			}
+
+			if drop.InterfaceBlock.OutDrops != nil && drop.InterfaceBlock.GetOutDrops() > 0 {
+				t.Errorf("ERROR: InDrops counter got %d, want 0", drop.InterfaceBlock.GetOutDrops())
+			}
+
+			if drop.InterfaceBlock.Oversubscription != nil && drop.InterfaceBlock.GetOversubscription() > 0 {
+				t.Errorf("ERROR: InDrops counter got %d, want 0", drop.InterfaceBlock.GetOversubscription())
+			}
+
```

Comment:
Delete blank line

Reply:
Thanks for checking. Removed extra blank line.
```go
                               t.Errorf("ERROR: InDrops counter got %d, want 0", drop.InterfaceBlock.GetOversubscription())
			}
		}
```